kubernetes/README.md

```diff
@@ -0,0 +1,182 @@
+
+### Nodepool lifecycle
+
+Certain features of node pools are immutable (e.g. machine type); to change
+such parameters requires creating a new node pool with the desired new values,
+migrating workloads off of the old node pool, and then deleting the old node pool.
+The node pool lifecycle scripts help simplify this process.
+
+#### Create a new node pool
+
+Configure a new node pool by adding its name to the `GKE_NODEPOOL_NAMES` array in [kubernetes/gke/config-nodepool.sh](./gke/config-nodepool.sh). For each nodepool property (`GKE_NODEPOOL_NODE_COUNT`, `GKE_NODEPOOL_NODE_LOCATIONS`, etc) it is required to add an entry to the array which is mapped to the nodepool name. This config file is also where you will set Kubernetes [taints](https://kubernetes.io/docs/concepts/scheduling-eviction/taint-and-toleration/) and [labels](https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/) on the nodes.
+
+Once the new nodepool is configured, it can be stood up by running `kubernetes/gke/nodepool-up.sh <nodepool-name>`, or by simply running `kubernetes/gke/nodepool-up.sh`, which will stand up all configured node pools which do not yet exist.
+
+#### Drain and delete an old node pool ####
+
+Once a new nodepool has been created to replace an active node pool, the old node pool must be removed from the `GKE_NODEPOOL_NAMES` array.
+
+Once the old node pool is removed from the array, it can be drained and deleted by running `kubernetes/gke/nodepool-down.sh <nodepool-name>`.
+
+## Deploying workloads
+
+Cluster workloads are divided into two classes:
+
+1. Apps are the workloads that users actually care about; this includes deployed "applications" such as the GTFS-RT archiver but also includes "services" like Grafana and Sentry. These workloads are deployed using `invoke` as defined in the [ci](../ci/) folder.
+2. System workloads are used to support running applications. This includes items such as an ingress controller, HTTPS certificate manager, etc. The system deploy command is run at cluster create time, but when new system workloads are added it may need to be run again.
+
+    ```bash
+    kubectl apply -k kubernetes/system
+    ```
+
+## JupyterHub
+
+JupyterHub is a good example of an application using a Helm chart that is ultimately exposed to the outside internet for user access. In general, any non-secret changes to the chart can be accomplished by modifying the chart's `values.yaml` and running the `invoke release` specific to JupyterHub.
+```
+poetry run invoke release -f channels/prod.yaml --app=jupyterhub
+```
+
+### Secrets
+Because we use [Github OAuth](https://zero-to-jupyterhub.readthedocs.io/en/latest/administrator/authentication.html?highlight=oauth#github) for user authentication in JupyterHub, we have to provide a client-id and client-secret to the JupyterHub Helm chart. Here is what the full configuration for the GitHub OAuth in our JupyterHub Helm chart's `values.yaml` might look like:
+
+```yaml
```

Comment:
did you want to replace this with a link to the real file?

Reply:
I think no, because technically this is the entire values.yaml structure that we split into 2 places